joomla/joomla-cms
# [com_installer] [updatesites] Add options to delete and rebuild update sites

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/8512.
#### Summary of Changes

After updating Joomla and extensions a lot of times (since 1.7) we get a lot of obsolete update sites from domain changes, url changes, etc that extension developers forget to delete on uninstall.

This makes searching for updates more slow. Right now we can disable the update sites, but we cannot delete or even rebuild the update sites table.

This PR adds two new options to com_installer updates sites view: delete and rebuild
- delete: deletes an update site (joomla core updates sites cannot be deleted)
- rebuild: the update sites are rebuild from the manifest files.
#### Testing Instructions
1. Use latest staging
2. Apply this patch
3. Go to "Extensions -> Manage -> Update Sites"
4. Notice the two new buttons
   ![image](https://cloud.githubusercontent.com/assets/9630530/14292445/06b6b824-fb60-11e5-9155-609a32a0f4e9.png)
5. Try to Rebuild the update sites and check if everything is fine.
6. Install some extensions with update sites of all the types you can imagine (libraries, templates, plugin, modules, components, etc).
7. Repeat step 5 and check if everything is fine.
8. Delete some update sites (the 4 core joomla update sites will not be allowed to delete) the others you can delete.
9. Repeat step 5 and check if everything is fine.
#### Observations

Code revisions, improvements and suggestions are welcomed.
